vm/object.go

```diff
@@ -22,26 +22,30 @@ func (p *Pointer) returnClass() *RClass {
 
 // RObject represents any non built-in class's instance.
 type RObject struct {
-	Class             *RClass
+	*baseObj
+	InitializeMethod *MethodObject
+}
+
+type baseObj struct {
```

Comment:
And from now on,  if objects' function use same attribute we just need to implement it on `baseObj` once. Just like `instanceVariableGet` and `instanceVariableSet`